gadgets/deadlock/go/go.mod

```
@@ -0,0 +1,9 @@
+module deadlock
+
+go 1.22.7
```

Since https://github.com/inspektor-gadget/inspektor-gadget/pull/3589#issuecomment-2416111765 we require 1.22.8:

```suggestion
go 1.22.8
```

The PR also needs a rebase.

---

@alban Done!